ext/nnstreamer/extra/nnstreamer_grpc_flatbuf.cc

```diff
@@ -29,56 +27,59 @@
 
 #include <gst/base/gstdataqueue.h>
 
-using nnstreamer::flatbuf::TensorService;
-using nnstreamer::flatbuf::Tensors;
-using nnstreamer::flatbuf::Tensor;
-using nnstreamer::flatbuf::Empty;
-
 using nnstreamer::flatbuf::Tensor_type;
 using nnstreamer::flatbuf::frame_rate;
 
-using flatbuffers::grpc::Message;
 using flatbuffers::grpc::MessageBuilder;
 
 using namespace grpc;
 
-/** @brief constructor */
-ServiceImplFlatbuf::ServiceImplFlatbuf (const grpc_config * config):
-  NNStreamerRPC (config), client_stub_ (nullptr)
+/** @brief Constructor of ServiceImplFlatbuf */
+ServiceImplFlatbuf::ServiceImplFlatbuf (const grpc_config * config)
+  : NNStreamerRPC (config), client_stub_ (nullptr)
 {
 }
 
-/** @brief client-to-server streaming: a client sends tensors */
-Status
-ServiceImplFlatbuf::SendTensors (ServerContext *context,
-    ServerReader<Message<Tensors>> *reader,
-    Message<Empty> *replay)
+/** @brief parse tensors and deliver the buffer via callback */
+void
+ServiceImplFlatbuf::parse_tensors (Message<Tensors> &tensors)
 {
-  return _read_tensors (reader);
+  GstBuffer *buffer;
+
+  _get_buffer_from_tensors (tensors, &buffer);
+
+  if (cb_)
```

Comment:
if cb is null, buffer should be released.
how about setting gst-buffer in if condition?
```
if (cb) {
  GstBuffer *buffer = get_buffer_from_tensor()
  cb (...)
}
```

Reply:
Also plz check the implementation in grpc-protobuf.
IMO others are ok.

Reply:
Oh, thanks. I will update.